From 142d4d10026f5b7ad847fd0f84c1b3bc2587885b Mon Sep 17 00:00:00 2001
From: Louis-Alexis Eyraud <louisalexis.eyraud@collabora.com>
Date: Mon, 24 Nov 2025 13:27:47 +0100
Subject: [PATCH 76/79] arm64: dts: mediatek: mt8365-evk: Fix touchscreen reset
 line polarity

Commit 7363096a5a08 ("Input: goodix - remove setting of RST pin to input")
modified the Goodix driver so it no longer tries to set reset pin as
input.

It led the driver to fail to probe during boot time on the Mediatek
Genio 350-EVK board because the reset pin polarity is incorrectly
declared in the board devicetree:
```
[    6.253819] Goodix-TS 1-005d: Error reading 1 bytes from 0x8140: -6
[    6.277884] Goodix-TS 1-005d: Error reading 1 bytes from 0x8140: -6
[    6.305503] Goodix-TS 1-005d: I2C communication failure: -6
```

so fix in the mt8365-evk.dts the reset-gpios property and pinctrl
configuration of the touchscreen node to use the proper polarity value.

Fixes: 7363096a5a08 ("Input: goodix - remove setting of RST pin to input")
Fixes: b28c4af8e44b ("arm64: dts: mt8365-evk: Add goodix touchscreen support")
Signed-off-by: Louis-Alexis Eyraud <louisalexis.eyraud@collabora.com>
---
 arch/arm64/boot/dts/mediatek/mt8365-evk.dts | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/mediatek/mt8365-evk.dts b/arch/arm64/boot/dts/mediatek/mt8365-evk.dts
index af43c392d3a0..6ede6a72525c 100644
--- a/arch/arm64/boot/dts/mediatek/mt8365-evk.dts
+++ b/arch/arm64/boot/dts/mediatek/mt8365-evk.dts
@@ -353,7 +353,7 @@ touchscreen@5d {
 		pinctrl-names = "default";
 		pinctrl-0 = <&touch_pins>;
 		irq-gpios = <&pio 78 GPIO_ACTIVE_HIGH>;
-		reset-gpios = <&pio 79 GPIO_ACTIVE_LOW>;
+		reset-gpios = <&pio 79 GPIO_ACTIVE_HIGH>;
 		AVDD28-supply = <&touch0_fixed_3v3>;
 		VDDIO-supply = <&mt6357_vrf12_reg>;
 	};
@@ -704,7 +704,7 @@ ctp-int1-pins {
 
 		rst-pins {
 			pinmux = <MT8365_PIN_79_CMVSYNC__FUNC_GPIO79>;
-			output-low;
+			output-high;
 		};
 	};
 
-- 
2.43.0

